### PR TITLE
XWIKI-21725: Cloudflare reject review requests without a user agent

### PR DESCRIPTION
**Pull request by manuelleduc**
Jira: https://jira.xwiki.org/browse/XWIKI-21725

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/java/org/xwiki/extension/security/internal/DefaultReviewsFetcher.java [hunk 1]
```diff
@@ -62,25 +61,19 @@ public class DefaultReviewsFetcher implements ReviewsFetcher
     @Inject
     private Logger logger;
 
+    @Inject
+    private HttpClientFactory httpClientFactory;
```
Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/java/org/xwiki/extension/security/internal/DefaultReviewsFetcher.java:65]: We should introduce a new, more generic, factory (this one is quite ExtensionManager centric) but that should do it for now.

**Review by tmortagne — APPROVED** [review]

File: xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/java/org/xwiki/extension/security/internal/DefaultReviewsFetcher.java [hunk 2]
```diff
@@ -62,25 +61,19 @@ public class DefaultReviewsFetcher implements ReviewsFetcher
...
-            .GET()
-            .build();
-
-        try {
-            HttpResponse<String> response = HttpClient.newHttpClient().send(request, ofString());
-            ReviewsMap reviewsMap = new ObjectMapper().readValue(response.body(), ReviewsMap.class);
+        try (CloseableHttpClient httpClient = this.httpClientFactory.createHttpClientBuilder(Map.of()).build()) {
+            HttpGet getMethod = new HttpGet(buildURI());
+            CloseableHttpResponse execute = httpClient.execute(getMethod);
+            ReviewsMap reviewsMap = new ObjectMapper().readValue(execute.getEntity().getContent(), ReviewsMap.class);
```
Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/java/org/xwiki/extension/security/internal/DefaultReviewsFetcher.java:73]: You might want to check if the request succeeded before reading it.

File: xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/java/org/xwiki/extension/security/internal/DefaultReviewsFetcher.java (hunk 1 above)
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/java/org/xwiki/extension/security/internal/DefaultReviewsFetcher.java:70]: done

File: xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/java/org/xwiki/extension/security/internal/DefaultReviewsFetcher.java (hunk 2 above)
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/java/org/xwiki/extension/security/internal/DefaultReviewsFetcher.java]: done

**Pull request merged**